### PR TITLE
common-updater-scripts: Support SRI-style hash

### PR DESCRIPTION
**Pull request by jtojnar**
Some fetcher functions support SRI-style `hash` attribute in addition to legacy type-specific attributes. When `hash` is used `outputHashAlgo` is null so let’s complain when SRI-style hash value was not detected.

Such attributes match the form `${type}${separator}${hash}`: True SRI uses dash as a separator and only supports base64, whereas Nix’s SRI-style format uses a colon and supports all the same encodings like regular hashes (16/32/64).

To keep this program reasonably simple, we will upgrade Nix’s SRI-like format to pure SRI instead of preserving it.

Relevant code:
* https://github.com/NixOS/nixpkgs/commit/267c8d6b2fea05bc811c8e2c2f4529b1436eeb9a
* https://github.com/NixOS/nix/blob/50d483a2c1c77bef1bab742eb8708626a61dc00e/src/libutil/hash.cc#L128-L206

### CODE REVIEW

**Comment by FRidh** [comment]
Is SRI already allowed in Nixpkgs? If I am correct it's a Nix `fetchurl` feature.

**Comment by jtojnar** [comment]
See https://github.com/NixOS/nixpkgs/commit/267c8d6b2fea05bc811c8e2c2f4529b1436eeb9a

File: pkgs/common-updater/scripts/update-source-version [hunk 1]
```diff
@@ -132,7 +132,19 @@ if [ -n "$newUrl" ]; then
...
 
+if [[ "$oldHash" =~ ^(sha256|sha512)[:-] ]]; then
+    # Handle the possible SRI-style hash attribute (in the form ${type}${separator}${hash})
+    # True SRI uses dash as a separator and only supports base64, whereas Nix’s SRI-style format uses a colon and supports all the same encodings like regular hashes (16/32/64).
+    # To keep this program reasonably simple, we will upgrade Nix’s format to SRI.
+    oldHashAlgo="${BASH_REMATCH[1]}"
+    sri=true
+elif [[ "$oldHashAlgo" = "null" ]]; then
+    # Some fetcher functions support SRI-style `hash` attribute in addition to legacy type-specific attributes. When `hash` is used `outputHashAlgo` is null so let’s complain when SRI-style hash value was not detected.
+    die "Unable to figure out hashng scheme from '$oldHash' in '$attr'!"
```
Comment (worldofpeace) [inline pkgs/common-updater/scripts/update-source-version]:
```suggestion
    die "Unable to figure out hashing scheme from '$oldHash' in '$attr'!"
```

Comment (worldofpeace) [reply pkgs/common-updater/scripts/update-source-version]: does the hard quoting here make it literally `$oldHash` in the error message?

Comment (jtojnar) [reply pkgs/common-updater/scripts/update-source-version]: Nope, the string literal type is determined by the outer quotes.

**Comment by worldofpeace** [comment]
Thanks for adding this @jtojnar

**Pull request merged**